packages/strapi-plugin-upload/services/Upload.js

```diff
@@ -11,14 +11,14 @@ const path = require('path');
 const crypto = require('crypto');
 const _ = require('lodash');
 const util = require('util');
-const filenamify = require('filenamify');
+const { nameToSlug } = require('strapi-utils');
 const mime = require('mime-types');
 
 const { bytesToKbytes } = require('../utils/file');
 
 const randomSuffix = () => crypto.randomBytes(5).toString('hex');
 const generateFileName = name => {
-  const baseName = filenamify(name, { replacement: '_' }).replace(/\s/g, '_');
+  const baseName = nameToSlug(name, { separator: '_', lowercase: false });
```

Comment:
Can you verify that filenamify is not required when using slugify ? no specific chars that could be a pbl in file names ? e.g `/` `\`

Reply:
Yes, I added all reserved characters from filenamify to the test except the `/`, as I don't think think is a possible input because it cannot be part of a file name either on Windows or on Unix-like systems. Furthermore, when I included it in the test, the `generateFileName` function would already get the wrong baseName from the `formatFileInfo` function.